src/app/api/vote/route.ts

```diff
@@ -0,0 +1,153 @@
+import { NextResponse } from "next/server";
+import { createClient } from "@supabase/supabase-js";
+import { decodeJwt } from "jose";
+
+interface bodyRequest {
+  matchId: string;
+  botChosen: string;
+  amount: number;
+  jwt: string;
+}
+
+interface jwtUserData {
+  id: string;
+  name: string;
+  email: string;
+  iat: number;
+  exp: number;
+}
+
+interface userData {
+  id: string;
+  name: string;
+  email: string;
+  tokens: number;
+  created_at: string;
+}
+
+interface matchData {
+  id: string;
+  challonge_match_id: number;
+  bot1: string;
+  bot2: string;
+  winner: string;
+  score_bot1: number;
+  round: number;
+  state: string;
+  start_time: string;
+  score_bot2: number;
+  ordering: number;
+  tournament_id: string;
+  updated_time: string;
+}
+
+export async function POST(request: NextResponse) {
+  const supabase = createClient(
+    process.env.DB_URL as string,
+    process.env.DB_SECRET_KEY as string
+  );
+
+  const body = await request.json();
```

Comment:
```suggestion
  let body: Body;
  try {
    body = await request.json();
  } catch {
    return NextResponse.json({ error: "Invalid JSON" }, { status: 400 });
  }
```
lets do a check on the request as well